### PR TITLE
Various fixes for compiler warnings

### PR DESCRIPTION
**Pull request by firscity**

### CODE REVIEW

**Review by xakep-amatop — CHANGES_REQUESTED** [review]

**Comment by xakep-amatop** [comment]
review has been finished, 
I think that I have found a few minor issues and have one major question
pls, give me an answer to the major question

File: xen-dom-mgmt/src/xen-dom-mgmt.c [hunk 1]
```diff
@@ -620,11 +622,13 @@ struct xen_domain *domid_to_domain(uint32_t domid)
 {
 	struct xen_domain *iter;
 
+	k_mutex_lock(&dl_mutex, K_FOREVER);
 	SYS_DLIST_FOR_EACH_CONTAINER (&domain_list, iter, node) {
 		if (iter->domid == domid) {
 			return iter;
```
[the remark this answers is not in the document]
Comment (firscity) [reply xen-dom-mgmt/src/xen-dom-mgmt.c:629]: Thanks, fixed.

File: xenstore-srv/src/xenstore_srv.c [hunk 2]
```diff
@@ -58,13 +60,19 @@ struct xs_entry *key_to_entry(const char *key)
 	char *tok, *tok_state;
 	struct xs_entry *next, *iter = NULL;
 	sys_dlist_t *inspected_list = &root_xenstore.child_list;
+	char key_buffer[XENSTORE_ABS_PATH_MAX + 1];
 	size_t keyl = strlen(key);
```
[the remark this answers is not in the document]
Comment (firscity) [reply xenstore-srv/src/xenstore_srv.c]: Yes, it is called from exported functions (xss handlers). Check was added.

File: xenstore-srv/src/xenstore_srv.c [hunk 3]
```diff
@@ -58,13 +60,19 @@ struct xs_entry *key_to_entry(const char *key)
...
 
+	if (keyl > XENSTORE_ABS_PATH_MAX)
+		return NULL;
+
 	if (strncmp(rootdir, key, keyl) == 0) {
 		return &root_xenstore;
 	}
 
-	for (tok = strtok_r(key, "/", &tok_state); tok != NULL; tok = strtok_r(NULL, "/", &tok_state)) {
+	strncpy(key_buffer, key, keyl);
```
[the remark this answers is not in the document]
Comment (firscity) [reply xenstore-srv/src/xenstore_srv.c]: Fixed, thanks.

File: xenstore-srv/src/xenstore_srv.c [hunk 4]
```diff
@@ -34,6 +34,8 @@ struct message_handle {
 	void (*h)(struct xen_domain *domain, uint32_t id, char *payload, uint32_t sz);
 };
 
+void xs_evtchn_cb(void *priv);
```
[the remark this answers is not in the document]
Comment (firscity) [reply xenstore-srv/src/xenstore_srv.c]: Added TODO related to this declaration. It is caused by some unusual usage of evtchn cb and should be fixed.

**Review by xakep-amatop — APPROVED** [review]

**Comment by xakep-amatop** [comment]
`Acked-by: Mykola Kvach <mykola_kvach@epam.com>`

File: xen-dom-mgmt/src/xen-dom-mgmt.c [hunk 5]
```diff
@@ -620,11 +620,14 @@ struct xen_domain *domid_to_domain(uint32_t domid)
 {
```
Comment (lorc) [inline xen-dom-mgmt/src/xen-dom-mgmt.c:622]: Issue with locking here is deeper than just locking `&domain_list`.

As I see,  typical usage is something like this:

```
	domain = domid_to_domain(domid);
	if (!domain) {
		printk("No domain with domid = %u is present\n", domid);
```

Problem is that `domain` can be freed/destroyed right after `domid_to_domain()` return. You either need to refcount domains or to globally lock all domains state.

Comment (firscity) [reply xen-dom-mgmt/src/xen-dom-mgmt.c:628]: Yes, I agree. Currently I replaced this changes in fixup with TODO, that contains problem description. I will analyze and fix it in separate PR. Thanks.

**Comment by LKomaryanskiy** [comment]
@firscity Good job! LGTM +1, you can add
`Acked-by: Leonid Komarianskyi <leonid_komarianskyi@epam.com>`

**Review by LKomaryanskiy — APPROVED** [review]

**Comment by lorc** [comment]
`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Pull request merged**